File: src/main/rst/examples/Chapter4/python/ProxySetupFirefox.py
from selenium import webdriver
from selenium.webdriver.common.proxy import Proxy

myProxy = "host:8080"

proxy = Proxy({
    # Direct = 0, Manual = 1, PAC = 2, AUTODETECT = 4, SYSTEM = 5
    'proxyType': 1,
    'httpProxy': myProxy,
    'ftpProxy': myProxy,
    'sslProxy': myProxy,
    'noProxy': '' # set this value as desired
    })

driver = webdriver.Firefox(proxy=proxy)

# for remote
caps = webdriver.DesiredCapabilities.FIREFOX
proxy.add_to_capabilities(caps)

driver = webdriver.Remote(desired_capabilities=caps)

File: src/main/rst/examples/Chapter4/ruby/ProxySetupFirefox.rb
require 'rubygems'
require 'selenium-webdriver'

PROXY = 'localhost:8087'

profile = Selenium::WebDriver::Firefox::Profile.new
profile.proxy = Selenium::WebDriver::Proxy.new(
  :http => PROXY,
  :ftp => PROXY,
  :ssl => PROXY,
  :no_proxy => nil
)

driver = Selenium::WebDriver.for :firefox, :profile => profile
